feat/search-equipment-and-medicine

lukapetrovic3110 opened the pull request.

lukapetrovic3110 committed:
refactor: graphical classes

lukapetrovic3110 committed:
feat: add search for equipment and medicine

lukapetrovic3110 committed:
feat: added window for search available appointment

lukapetrovic3110 committed:
fix: code smells

LukaKrickovic commented on HesoyamHospital/GraphicEditor/Service/SearchService.cs:
```diff
@@ -8,48 +7,74 @@ class SearchService
-            List<MapLocation> mapLocation3 = Find_in_file("hospital2groundfloor.txt", name);
-            List<MapLocation> mapLocation4 = Find_in_file("hospital2firstfloor.txt", name);
-            List<MapLocation> mapLocation5 = Find_in_file("hospital2secondfloor.txt", name);
+            List<MapLocation> mapLocation1 = ReadObjects("hospital1groundfloor.txt", name);
```
Maybe try refactoring this in to a for loop, to reduce line duplication

nidza0 commented on HesoyamHospital/GraphicEditor/Model/EquipmentAndMedicine.cs:
```diff
@@ -0,0 +1,28 @@
+﻿
+namespace GraphicEditor
```
Is this a DTO object or a model?
If it's a model, it should be placed within a Backend project.
Contact Web team to check if they are already working on something similiar to reduce a chance for conflict and time loss.

nidza0 commented on HesoyamHospital/GraphicEditor/Repository/EquipmentAndMedicineRepository.cs:
```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.IO;
+
+namespace GraphicEditor
```
Same thing here. Repository should be within a Backend project.
We want to use the same model and repository to be used from every application, maintaining consistency.

nidza0 commented on HesoyamHospital/GraphicEditor/View/Search.xaml.cs:
```diff
@@ -24,9 +25,27 @@ public Search()
+            {
+                if (name.IsNullOrEmpty())
+                {
+                    List<EquipmentAndMedicine> results = searchService.FindAllEquipmentAndMedicines("equipmentandmedicine.txt");
```
Equipment and medicine should be placed within a database. If you are not familiar with how adding a new entity to database work, contact a member from Devops team or Web team to get introduced to how ORM works.

nidza0 commented:
Take a look at above suggestions and let me know what you think.

nidza0 reviewed (dismissed):
.

milica1612 committed:
refactor: refactored search service

milica1612 committed:
Merge branch 'develop' into feat/search-equipment-and-medicine

milica1612 committed:
fix: connecting iventory item to database

lukapetrovic3110 committed:
refactor: type of the room

EXAMINATIONROOM -> EXAMINATION

milica1612 committed:
refactor: refactor DTO from medicine and equipment

lukapetrovic3110 committed:
refactor: ComboBox for Search

removed ComboBoxItem searchEquipmentAndMedicine
added ComboBoxItem searchEquipment
added ComboBoxItem searchMedicine

lukapetrovic3110 committed:
feat: added MedicineDTO

lukapetrovic3110 committed:
feat: added EquipmentDTO

lukapetrovic3110 committed:
fix: code smell

lukapetrovic3110 committed:
feat: search for medicines

milica1612 committed:
feat: search equipment

milica1612 committed:
feat:advanced equipment search

lukapetrovic3110 committed:
refactor: changed type of  a variable room in MedicineDTO

lukapetrovic3110 committed:
Merge branch 'feat/search-equipment-and-medicine' of https://github.com/PSW-2020-ORG2/Hesoyam-Hospital into feat/search-equipment-and-medicine

lukapetrovic3110 committed:
feat: advanced search of medicines

lukapetrovic3110 committed:
fix: code smells

lukapetrovic3110 committed:
fix: code smells

lukapetrovic3110 committed:
fix: code smells

lukapetrovic3110 committed:
chore: remove unnecessary using

lukapetrovic3110 committed:
Merge branch 'develop' into feat/search-equipment-and-medicine
